h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -0,0 +1,139 @@
+var setElementState = require('../util/dom').setElementState;
+
+/**
+ * Controller for the search bar.
+ */
+class SearchBarController extends Controller {
+  constructor(element) {
+    super(element);
+
+    this._input = element.querySelector('.js-search-bar-input');
+    this._dropdown = element.querySelector('.js-search-bar-dropdown');
+    this._dropdownItems = element.querySelectorAll('.js-search-bar-dropdown-item');
+
+    var self = this;
+    var index = -1;
+    var menuLen = this._dropdownItems.length;
+
+    var updateActiveDropdownItem = function (index) {
+      clearActiveDropdownItem();
+      self._dropdownItems[index].classList.add('js-search-bar-dropdown-menu-item--active');
+    };
+
+    var clearActiveDropdownItem = function () {
+      if (element.querySelector('.js-search-bar-dropdown-menu-item--active')) {
+        element.querySelector('.js-search-bar-dropdown-menu-item--active').
+        classList.remove('js-search-bar-dropdown-menu-item--active');
+      }
+    };
+
+    var onOpenDropdown = function () {
+      self.setState({open: true});
+      self._input.addEventListener('keydown', setupListenerKeys,
+        true /*capture*/);
+    };
+
+    var onCloseDropdown = function () {
+      index = -1;
+      clearActiveDropdownItem();
+      self.setState({open: false});
+      self._input.removeEventListener('keydown', setupListenerKeys,
+        true /*capture*/);
+    };
+
+    var onSelectFacet = function (facet) {
+      self._input.value = self._input.value + facet;
+
+      onCloseDropdown();
+
+      setTimeout(function() {
+        self._input.focus();
+      }, 0);
+    }
+
+    var setupListenerKeys = event => {
+      switch (event.keyCode) {
+      case 13:
+        // enter key
+        event.preventDefault();
+
+        if (element.querySelector('.js-search-bar-dropdown-menu-item--active')) {
+          var facet = element.querySelector('.js-search-bar-dropdown-menu-item--active').
+            querySelector('.js-search-bar-dropdown-item-title').innerHTML.trim();
+          onSelectFacet(facet);
+        }
+        break;
+      case 38:
+        // up key
+        if (index < 0) {
+          index = index + 1;
+        }
+        index = 
+          (((index - 1) % menuLen) + menuLen) % menuLen;
+        updateActiveDropdownItem(index);
+        break;
+      case 40:
+        // down key
+        index = (index + 1) % menuLen;
+        updateActiveDropdownItem(index);
+        break;
+      default:
+        break;
+      };
+    };
+
+    var handleClickOnItem = event => {
+      var facet = event.currentTarget.querySelector('.js-search-bar-dropdown-item-title').innerHTML.trim();
+      onSelectFacet(facet);
+    }
+
+    var handleHoverOnItem = event => {
+      clearActiveDropdownItem();
+      index = Array.prototype.slice.call(this._dropdownItems).indexOf(event.currentTarget)
+      event.currentTarget.classList.add('js-search-bar-dropdown-menu-item--active');
```

Comment:
I think this should call `updateActiveDropdownItem()` so that there's only one function that adds and removes the `'--active'` class